lib/LoginStrategy.ts

```diff
@@ -256,6 +275,49 @@ export class LoginStrategy extends BaseStrategy {
       return done(error);
     }
 
+    return await this.getVerifiedAcceptedRoles(userDid, done);
+  }
+
+  /**
+   * Verifies Siwe Message and signature
+   * @param token signature
+   * @param payload Siwe payload
+   * @param done
+   * @returns
+   */
+  private async verifySiweToken(
+    token: string,
+    payload: Partial<SiweMessagePayload>,
+    done: (err?: Error, user?: unknown, info?: unknown) => void
+  ): Promise<void> {
+    const userDid = this.didUnification(`did:ethr:${payload.address}`);
+    const siwe = new SiweMessage(payload);
+    try {
+      await siwe.verify({
+        signature: token,
+        domain: payload.domain,
+        nonce: payload.nonce,
+        time: payload.issuedAt,
+      });
+    } catch (error) {
+      const err = error as SiweResponse;
+      const errorMessage = err.error?.type;
+      return done(new Error(errorMessage));
+    }
+    return this.getVerifiedAcceptedRoles(userDid, done);
+  }
```

Comment:
This verification should also check the `uri` value of the payload as I think this check is not being performed by the `siwe` library, I believe.
I think the expected `uri` should be set in the `LoginStrategy` constructor.
This check is important in order to help prevent replay attacks.